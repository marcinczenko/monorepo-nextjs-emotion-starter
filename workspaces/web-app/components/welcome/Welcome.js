import React, { useState } from 'react'
import { PageCentered, Centered, Spacer } from '@react-frontend-developer/react-layout-helpers'
import { Segment, Button, Icon, Header } from 'semantic-ui-react'

const Welcome = () => {
  const [ iconName, setIconName ] = useState('thumbs up')

  const onClick = () => {
    iconName === 'thumbs up'
      ? setIconName('thumbs up outline')
      : setIconName('thumbs up')
  }

  return (
    <PageCentered>
      <Segment raised css={{ width: '80%', minHeight: '254px', maxWidth: '500px' }}>
        <Centered>
          <Header as='h1'>Welcome</Header>
          <Icon name={iconName} size='massive' />
          <Spacer margin='30px 0 0 0'>
            <Button onClick={onClick} basic color='black'>Click Me!</Button>
          </Spacer>
        </Centered>
      </Segment>
    </PageCentered>
  )
}

export { Welcome }
