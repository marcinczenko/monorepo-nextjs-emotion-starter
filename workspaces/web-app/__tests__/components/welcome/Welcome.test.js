import React from 'react'
import { render } from 'react-testing-library'
import { Welcome } from '../../../components/welcome'

describe('Welcome', () => {
  it('has proper styling', () => {
    const { container } = render(<Welcome />)
    expect(container).toMatchSnapshot()
  })

  it('displays welcome text', () => {
    const { getByText } = render(<Welcome />)
    expect(getByText(/welcome/i)).toBeInTheDocument()
  })
})
