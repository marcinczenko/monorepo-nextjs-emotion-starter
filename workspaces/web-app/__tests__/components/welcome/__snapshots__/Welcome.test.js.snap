// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome has proper styling 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100vh;
  width: 100vw;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="ui raised segment"
      css="[object Object]"
    >
      <div
        class="emotion-0 emotion-1"
      >
        <h1
          class="ui header"
        >
          Welcome
        </h1>
        <i
          aria-hidden="true"
          class="thumbs up massive icon"
        />
        <div
          style="margin: 30px 0px 0px 0px; padding: 0px; box-sizing: border-box;"
        >
          <button
            class="ui black basic button"
          >
            Click Me!
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Welcome show the thumbs up icon again when button is pressed second time 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100vh;
  width: 100vw;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="ui raised segment"
      css="[object Object]"
    >
      <div
        class="emotion-0 emotion-1"
      >
        <h1
          class="ui header"
        >
          Welcome
        </h1>
        <i
          aria-hidden="true"
          class="thumbs up massive icon"
        />
        <div
          style="margin: 30px 0px 0px 0px; padding: 0px; box-sizing: border-box;"
        >
          <button
            class="ui black basic button"
          >
            Click Me!
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Welcome show the thumbs up outline icon when button is pressed 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100vh;
  width: 100vw;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="ui raised segment"
      css="[object Object]"
    >
      <div
        class="emotion-0 emotion-1"
      >
        <h1
          class="ui header"
        >
          Welcome
        </h1>
        <i
          aria-hidden="true"
          class="thumbs up outline massive icon"
        />
        <div
          style="margin: 30px 0px 0px 0px; padding: 0px; box-sizing: border-box;"
        >
          <button
            class="ui black basic button"
          >
            Click Me!
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
